### DISCUSSION TITLE
Is it possible to use this in CMD in Windows OS?

### DISCUSSION THREAD

**Discussion by JenuelDev** [starship/starship#3041] [Q&A] [answered]
Is it possible to use this in CMD in Windows OS? I mostly use CMD in some projects that need CMD commands, hoping starship has support for it. Thanks :)

**Comment by chipbuster** [comment]
Unfortunately, this seems slightly tricky to do, not just for starship, but for any custom prompt + CMD.

The issue I can see is that the cmd `prompt` command [only allows text and fixed codes with special meanings](https://stackoverflow.com/a/31415574). It doesn't look like it allows things like variable expansions (which, from what I can tell, don't really exist within CMD like they exist on other shells). This means that it's not possible to hook starship into cmd the way we do it for all other shells.

All is not lost: the prompt command *does* allow you to expand environmental variables, so if we could figure out how to set `starship_text` every time the prompt draws, we could do something like `prompt %starship_text%` and have a working prompt. Unfortunately, I don't know how to do this in CMD. [This SO post](https://stackoverflow.com/questions/43210462/how-to-run-cmd-command-every-time-one-is-executed) has a comment suggesting that the `doskey` command may be of use here, but unfortunately I'm not familiar enough with it to be able to figure out how to make it work.

**Reply by JenuelDev** [reply]
Oh nice, thanks for the reply, I don't know if this will be a right answer, but well wait for more suggestions,, thanks for the Idea... :)

**Reply by chipbuster** [reply]
I don't want to lose my work, so here's a proof-of-concept.

starship.toml:

```
# Multiline variables are a total pain in cmd
[line_break]
disabled=true

# Can't use UTF-8, but need to escape `>` or its treated as a special character in prompt
[character]
success_symbol = "^>"
error_symbol = "^>"
```

![image](https://user-images.githubusercontent.com/4605384/133498765-f50ad9a7-b5a6-4faa-974e-60f6fc1d6e4b.png)

As you can see, I've set the prompt, but in order to update it, I need to run two commands:

```
for /f "tokens=* usebackq" %F in (`starship prompt -P %cd%`) do ( SET ssp=%F )
prompt %ssp%
```

Once I run those two, the prompt properly updates. If we could figure out how to auto-run these commands, we'd be in business.

**Comment by chipbuster** [comment]
@BroJenuel If you have the option to install CLink, we'll hopefully be merging cmd + CLink support in #3277 (it probably won't get merged for a few more days because I'm currently overwhelmed by PRs, but I'm hoping before next year!)

**Comment by chipbuster** [comment, marked as answer]
It will be possible to use this prompt with cmd.exe with CLink in the next major version of starship (1.2.0) now that #3277 has been merged.

**Reply by JenuelDev** [reply]
awesome!!
